libopenage/terrain/terrain_search.cpp

```diff
@@ -52,6 +75,7 @@ void TerrainSearch::reset() {
 	std::swap(this->tiles, empty);
 	this->visited.clear();
 	this->tiles.push(this->start);
+	this->visited.insert(this->start);
```

Comment:
Because next_tile only adds tiles to visited when it adds them to this->tiles, so it will add the start tile again on its second run. So we'd check the start tile twice which is a) kind of useless and b) makes it more difficult to detect when the search was reset.

Now that I'm thinking about it, I'm not sure if it even makes any sense to have this->tiles and this>visited since they seem to always contain the same elements.